### ISSUE TITLE
Has someting been changed with merging sequenced sources in cross?

### ISSUE THREAD

**Issue by JohanJNL** [savonet/liquidsoap#3847]
For quite some time I've tried to improve crossfading in Liquidsoap. Although I'm not a programmer, I still realize that my solution is very ugly. But the result is quite amazing. My script checks every second during the last seconds of a music file between sequence or crossfade .I wrote the script it in Liquidsoap 2.1.3 and 2.1.4.

After upgrading to 2.2.4 it stopped working. I tried to diagnose the reason and I think it's because now all sequenced sources are merged in cross, even with  sequence(merge=false, [a.source, b.source]).

I call the transition script in a while loop like this:

duration = ref(src)
 while !duration > 0 do
  src := cross(minimum=0., duration=float(!duration), width=(float(!duration) + 1.), transition, !src)
  duration := !duration - 1
 end

Because I cannot stop the while loop from the transition function, I have to stop the cross function in a different way, for example when the a.source is a jingle. I do that with sequence(merge=true, [a.source, b.source]). Since the upgrade to Liquidsoap 2.2.4 merge=true seems to happen always, or I I wrong?

I hope my issue is clear. I am not a native speaker. Anyway, thanks for offering Liquidsoap. I use it 24/7 on a stream and every Sunday for my broadcast via medium wave.

**Comment by gAlleb** [comment]
Hi!

As far as I can tell `the !x notation for getting the value of a reference is now deprecated. You should write x() instead. `

Also since that time a lot of new stuff has been implemented in regards to `crossfade`. See https://www.liquidsoap.info/blog/2024-02-07-liquidsoap-v2.2.4/ 

Also If you can upgrade to latest Rolling Release (https://github.com/savonet/liquidsoap/releases) you could take advantage of new `autocue` protocol and/or use `autocue2` option by @Moonbase59 here https://github.com/Moonbase59/autocue with some additional features. This would make possible for your to compute cue and/or fade values on the fly.

As for the time being new Rolling Releases should be as stable as ordinary ones.

**Comment by JohanJNL** [comment]
Thanks for your reply.

I know the !x notation is deprecated, but as long as I keep switching between 2.1.4 and newer releases, it's easier because it still works.

I don't see a options to cross at different times with the new features, so for now I continue using my script with the ugly while loop. I'm happy with the crossfades and it works in 2.1.4. I hope I can upgrade in near future.

**Comment by gAlleb** [comment]
Maybe you could try `single_track=false` instead of `merge=false`

**Comment by JohanJNL** [comment]
No, this doesn't solve the issue. First I tried it only with single_track=false, later together with merge=false. Both tests have the same result that I can't do a new cross with a lower duration and width.

**Comment by toots** [comment]
Hi @JohanJNL ! I'm not sure what the script is expected to do but I'm pretty sure it might have been working by chance!

What exactly are you trying to implement?
